kernel/yosys.cc

```diff
@@ -868,7 +868,10 @@ std::string proc_self_dirname()
 	return "/";
 }
 #else
-	#error "Don't know how to determine process executable base path!"
+std::string proc_self_dirname()
+{
+	return PREFIX "/bin/";
```

Comment:
The whole point of this function is to *avoid* hardcoding the directory, so that Yosys works anywhere. If you try to run the Yosys test suite (which implicitly assumes being run in the source directory), I am fairly sure it will fail because it cannot locate itself, and thus cannot locate any of the share directories that the synthesis passes rely on.

I googled around for a solution for openbsd, but I think you might need a hack to the effect of saving `argv[0]`, `getcwd()`, and `getenv("PATH")` and then attempt to locate the binary that way.

Reply:
This was what was done in the OpenBSD package, which is better than not having Yosys package at all, but for something upstream including the test suites...

So let's try with the argv trick, which is portable.